src/engine.cpp

```diff
@@ -737,7 +737,11 @@ void StartupOneEngine(Engine *e, const TimerGameCalendar::YearMonthDay &aging_ym
 	/* Don't randomise the start-date in the first two years after gamestart to ensure availability
 	 * of engines in early starting games.
 	 * Note: TTDP uses fixed 1922 */
-	e->intro_date = ei->base_intro <= TimerGameCalendar::ConvertYMDToDate(_settings_game.game_creation.starting_year + 2, 0, 1) ? ei->base_intro : (TimerGameCalendar::Date)GB(r, 0, 9) + ei->base_intro;
+	if (_settings_game.vehicle.vehicle_intro_randomisation) {
+		e->intro_date = ei->base_intro <= TimerGameCalendar::ConvertYMDToDate(_settings_game.game_creation.starting_year + 2, 0, 1) ? ei->base_intro : (TimerGameCalendar::Date)GB(r, 0, 9) + ei->base_intro;
+	} else {
+		e->intro_date = ei->base_intro;
+	}
```

Comment:
I'd clean up this logic to avoid the ternary, something like:

```suggestion
	TimerGameCalendar::Date begin_random_date = TimerGameCalendar::ConvertYMDToDate(_settings_game.game_creation.starting_year + 2, 0, 1);
	if (_settings_game.vehicle.vehicle_intro_randomisation && ei->base_intro > begin_random_date) {
		e->intro_date =  static_cast<TimerGameCalendar::Date>(GB(r, 0, 9)) + ei->base_intro;
	} else {
		e->intro_date = ei->base_intro;
	}
```